### PR TITLE
Feature -  RSetCache.addIfAbsent(Map<V, Duration> objects)  added

### PR DESCRIPTION
**Pull request by lyrric**
Added RSetCache.addIfAbsent(Map<V, Duration> objects) method

### CODE REVIEW

File: redisson/src/main/java/org/redisson/api/RSetCache.java [hunk 1]
```diff
@@ -136,6 +136,15 @@ public interface RSetCache<V> extends RSet<V>, RExpirable, RSetCacheAsync<V>, RD
...
     int addAllIfAbsent(Map<V, Duration> objects);
+    /**
+     * Adds elements to this set only if all of them haven't been added before.
+     * <p>
+     * Requires <b>Redis 3.0.2 and higher.</b>
+     *
+     * @param objects map of elements to add
+     * @return <code>true</code> if elements added and <code>false</code> if not.
+     */
+    boolean addAllIfAllAbsent(Map<V, Duration> objects);
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/api/RSetCache.java:147]: It's better to rename it to `addIfAbsent()`

File: redisson/src/main/java/org/redisson/api/RSetCacheAsync.java [hunk 2]
```diff
@@ -126,6 +126,15 @@ public interface RSetCacheAsync<V> extends RSetAsync<V> {
...
     RFuture<Integer> addAllIfAbsentAsync(Map<V, Duration> objects);
+    /**
+     * Adds elements to this set only if all of them haven't been added before.
+     * <p>
+     * Requires <b>Redis 3.0.2 and higher.</b>
+     *
+     * @param objects map of elements to add
+     * @return amount of added elements
+     */
+    RFuture<Boolean> addAllIfAllAbsentAsync(Map<V, Duration> objects);
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/api/RSetCacheAsync.java:137]: It's better to rename it to `addIfAbsentAsync()`

**Review by mrniko — CHANGES_REQUESTED** [review]

File: redisson/src/main/java/org/redisson/api/RSetCacheAsync.java (hunk 2 above)
Comment (lyrric) [reply redisson/src/main/java/org/redisson/api/RSetCacheAsync.java]: Thanks for reminding.

**Comment by lyrric** [comment]
All the problems have been solved, please check, thanks. @mrniko

File: redisson/src/main/java/org/redisson/RedissonSetCache.java [hunk 3]
```diff
@@ -1273,6 +1273,11 @@ public int addAllIfAbsent(Map<V, Duration> objects) {
         return get(addAllIfAbsentAsync(objects));
     }
 
+    @Override
+    public boolean addIfAbsent(Map<V, Duration> objects) {
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/RedissonSetCache.java:1277]: This method should be added RSetCacheReactive and RSetCacheRx interfaces as well.

**Review by mrniko — CHANGES_REQUESTED** [review]

File: redisson/src/main/java/org/redisson/RedissonSetCache.java (hunk 3 above)
Comment (lyrric) [reply redisson/src/main/java/org/redisson/RedissonSetCache.java:1277]: > This method should be added to RSetCacheReactive and RSetCacheRx interfaces as well.

OK，it has been added, please check.

**Comment by mrniko** [comment]
Please correct all checks

**Comment by lyrric** [comment]
> Please correct all checks

Sorry, fixed.

**Comment by mrniko** [comment]
Thanks for contribution!

**Pull request merged**